rust/src/utils.rs

```diff
@@ -453,3 +456,63 @@ pub(crate) fn get_rpm_basearch() -> String {
         s => s.to_string(),
     }
 }
+
+// also defined in rpmostree-rpm-util.h
+const GV_ADVISORY_TYPE_STR: &str = "(suuasa{sv})";
+// oddly, AFAICT there isn't an easy way to use GV_ADVISORIES_TYPE_STR in this definition
+const GV_ADVISORIES_TYPE_STR: &str = "a(suuasa{sv})";
+
+lazy_static::lazy_static! {
+    static ref GV_ADVISORY_TYPE: &'static glib::VariantTy = {
+        glib::VariantTy::new(GV_ADVISORY_TYPE_STR).unwrap()
+    };
+    static ref GV_ADVISORIES_TYPE: &'static glib::VariantTy = {
+        glib::VariantTy::new(GV_ADVISORIES_TYPE_STR).unwrap()
+    };
+}
+
+fn get_commit_advisories(repo: &mut ostree::Repo, checksum: &str) -> Result<Option<glib::Variant>> {
+    let (commit, _) = repo.load_commit(checksum)?;
+    let metadata = &variant_utils::variant_tuple_get(&commit, 0).expect("commit metadata");
+    let metadata = variant_utils::byteswap_be_to_native(metadata);
+    let dict = &glib::VariantDict::new(Some(&metadata));
+    Ok(dict.lookup_value("rpmostree.advisories", Some(&GV_ADVISORIES_TYPE)))
+}
+
+/// Compares advisories between two checksums and returns new ones in checksum_to ("diff" is a bit
+/// misleading here; we don't actually say which advisories were removed because normally we only
+/// really care about *new* advisories).
+pub(crate) fn calculate_advisories_diff(
+    mut repo: Pin<&mut crate::FFIOstreeRepo>,
+    checksum_from: &str,
+    checksum_to: &str,
+) -> Result<*mut crate::FFIGVariant> {
+    let mut repo = repo.gobj_wrap();
```

Comment:
Ultimately...I think higher level tooling is going to be oriented towards the JSON serialization.  Which argues for having this in something like coreos-assembler instead or possibly even higher level.

We may end up wanting to expose Go/Rust bindings for the JSON serialization of our GVariants...or perhaps just the schema.

Basically, I'm not objecting to shipping this code, but in practice I think it'll be easier to do it via a dynamic JSON query rather than pulling two commits into a a repo and running this command.

Reply:
This code is also needed for `rpm-ostree status`, so at that point it's just easier to make it accessible to outside users via `db diff --format json`. It's a straight shoe-in for cosa since it already calls that.